go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go

```diff
@@ -269,18 +270,22 @@ func (t *TxThrottler) Close() {
 // It returns true if the transaction should not proceed (the caller
 // should back off). Throttle requires that Open() was previously called
 // successfully.
-func (t *TxThrottler) Throttle() (result bool) {
+func (t *TxThrottler) Throttle(criticality int) (result bool) {
 	if !t.config.enabled {
 		return false
 	}
 	if t.state == nil {
 		panic("BUG: Throttle() called on a closed TxThrottler")
 	}
-	result = t.state.throttle()
+
+	// Throttle according to both what the throttle state says, and the criticality. Workloads with higher criticality
+	// are less likely to be throttled.
+	result = t.state.throttle() && rand.Intn(100) < 100-int(criticality)
```

Comment:
The magic number `100` is scattered a bit too much. Please create a `const` in a place that makes sense, name it something like `MaxCriticalityValue` or whatever name makes sense, than use that const throughout the code.

Reply:
`criticality` is already an `int`, there is no reason to cast it via `int(criticality)`

Reply:
Would it make more sense to invert the equation and simplify the condition? Like so:

```go
result = t.state.throttle() && rand.Intn(100) > criticality
```